CMakeLists.txt

```diff
@@ -23,7 +23,7 @@ if(UTF8_INSTALL)
         set(DEF_INSTALL_CMAKE_DIR CMake)
     else()
         include(GNUInstallDirs) # define CMAKE_INSTALL_*
-        set(DEF_INSTALL_CMAKE_DIR ${CMAKE_INSTALL_LIBDIR}/cmake/utf8cpp)
+        set(DEF_INSTALL_CMAKE_DIR ${CMAKE_INSTALL_DATADIR}/cmake/utf8cpp)
```

Comment:
Looking at the [documentation](https://cmake.org/cmake/help/latest/module/GNUInstallDirs.html) DATADIR is for `read-only architecture-independent data (DATAROOTDIR)` while LIBDIR is for `object code libraries (lib or lib64 or lib/<multiarch-tuple> on Debian)`. 

On Arch Linux LIBDIR points to /usr/lib while DATADIR points to /usr/share. The find_package [documentation](https://cmake.org/cmake/help/latest/command/find_package.html#search-procedure) says it would search in both of these places for the .cmake file. Since utfcpp is header only and has no other dependencies, I'm inclined to change this to DATADIR. On my particular Arch Linux install I have no examples of any installed projects installing to /usr/share/cmake other than gtk-doc while there are 237 individual packages that install to /usr/lib/cmake, but that certainly doesn't mean it's wrong. It should work either way.